docs/integration-guide.md

```diff
@@ -107,6 +107,319 @@ Here's how we recommend doing it:
+  function including any needed helper functions.
+
+### Walkthrough of `Hello World` webhook
+
+Below explains each part of a simple webhook integration, called **Hello
+World**. This webhook sends a "hello" message to the `test` stream and includes
+a link to the Wikipedia article of the day, which it formats from json data it
+receives in the http request.
+
+Use this walkthrough to learn how to write your first webhook
+integration.
+
+#### Step 0: Create fixtures
+
+The first step in creating a webhook is to examine the data that the
+service you want to integrate will be sending to Zulip.
+
+You can use [requestb.in](http://requestb.in/) or a similar tool to capture
+webook payload(s) from the service you are integrating. Examining this
+data allows you to do two things:
+
+1. Determine how you will need to structure your webook code, including what
+   message types your integration should support and how; and,
+2. Create fixtures for your webook tests.
+
+Fixtures enable the testing of webhook integration code without the need to
+actually contact the service being integrated.
+
+Because `Hello World` is a very simple webhook that does one thing, it requires
+only one fixture, `zerver/fixtures/helloworld/helloworld_hello.json`:
+
+```
+{
+  "featured_title":"Marilyn Monroe",
+  "featured_url":"https://en.wikipedia.org/wiki/Marilyn_Monroe",
+}
+```
+
+When writing your own webhook integration, you'll want to write one test
+function for every message type that your webhook supports. You'll also need a
+corresponding fixture for each of these tests. See [Step 3: Create
+tests](#step-3-create-tests) or [Testing](testing.html) for further details.
+
+#### Step 1: Create main webhook code
+
+The majority of the code for your webhook integration will be in a single
+python file in `zerver/views/webhooks/`.
+
+The Hello World integration is in `zerver/views/webhooks/helloworld.py`:
+
+```
+from __future__ import absolute_import
+from django.utils.translation import ugettext as _
+from zerver.lib.actions import check_send_message
+from zerver.lib.response import json_success, json_error
+from zerver.decorator import REQ, has_request_variables, api_key_only_webhook_view
+from zerver.lib.validator import check_dict, check_string
+
+@api_key_only_webhook_view('HelloWorld')
+@has_request_variables
+def api_helloworld_webhook(request, user_profile, client,
+                           payload=REQ(argument_type='body'),
+                           stream=REQ(default='test'),
+                           topic=REQ(default='Hello World')):
+
+  # construct the body of the message
+  body = ('Hello! I am happy to be here! :smile: ')
+
+  # try to add the Wikipedia article of the day
+  # return appropriate error if not successful
+  try:
+      body_template = '\nThe Wikipedia featured article for today is **[%s](%s)**'
+      body += body_template % (payload['featured_title'], payload['featured_url'])
```

Comment:
I personally preferred the `%` syntax, but `.format` is the new Python standard, so we should probably migrate towards it.  However, I think we should probably start using and recommending named placeholders with this sort of thing, e.g. for the variable part of this string, use something like this:

'[{featured_titled}]({featured_url})'.format(payload)

since that seems like a much cleaner way to use `.format()`.  What do you think @TomaszKolek ?